test/conftest.py

```diff
@@ -291,56 +287,168 @@ def pytest_collection_modifyitems(items: List[Any]) -> None:
 
 
 class StepcurrentPlugin:
-    # Modified fromo _pytest/stepwise.py in order to save the currently running
-    # test instead of the last failed test
+    """
+    This mean to work in with test/run_test.py to ensure that every test is run,
+    gets a retries in new subprocesses, and create xml.
+
+    Cache file contents:
+    pytest_previous_status: None on the first run, and then is either an exit
+    code or "no xml"
+    to_run: all tests that have still not created xml
+    prev_run: tests that are expected to run during this run.  Poorly named in
+    this context, but makes sense from test/run_test.py's POV.  If a test fails,
+    then it is possibly that not all tests in this actually ran. This should be
+    a prefix of to_run
+    already_ran: tests that have created xml
+
+    Test that are still to be run have 4 possible statuses, which are in
+    test/run_test.py and are: cont, s, r2, and r1, abbreviated c, s, 2, and 1 in
```

Comment:
Added to comment.  It would be better to use enums, but I'm not sure if importing between conftest and run_test is a good idea, so I'm leaving the statuses simple as possible to reduce probability of typos